Foundation/NSMeasurement.swift

```diff
@@ -23,21 +23,93 @@ open class NSMeasurement : NSObject, NSCopying, NSSecureCoding {
         self.unit = unit
     }
     
-    open func canBeConverted(to unit: Unit) -> Bool { NSUnimplemented() }
+    open func canBeConverted(to otherUnit: Unit) -> Bool {
+        #if DEPLOYMENT_RUNTIME_OBJC //|| os(Linux)
+            return otherUnit.isKind(of: type(of: unit))
+        #else
+            // just check conversion
+            if unit is Dimension && otherUnit is Dimension {
+                return true
+            } else {
+                return unit.isEqual(otherUnit)
+            }
+        #endif
+    }
     
-    open func converting(to unit: Unit) -> Measurement<Unit> { NSUnimplemented() }
+    open func converting(to otherUnit: Unit) -> Measurement<Unit> {
+        if canBeConverted(to: otherUnit) {
+            if unit.isEqual(otherUnit) {
+                return Measurement(value: doubleValue, unit: otherUnit)
+            } else {
+                guard let dimension = unit as? Dimension,
+                    let otherDimension = otherUnit as? Dimension else {
+                        fatalError("Cannot convert differing units that are non-dimensional! lhs: \(type(of: unit)) rhs: \(type(of: otherUnit))")
```

Comment:
Try to align these error messages as closely as possible with Darwin Foundation.